### ISSUE TITLE
MDF4 file with extended ID not useable in CANAlyzer

### ISSUE THREAD

**Issue by tsayman** [CSS-Electronics/mdf4-converters#2]
MDF4 file that has been finalized using mdffinalized.exe and contains extended ID data (e.g. J1939) cannot be run in offline mode in CANAlyzer.

System	Start of measurement 09:04:46.506 am
System	offline mode started
System	Error: Found CAN identifier (0x14FF2321) > 0x7ff without extended bit!
System	       Stopping measurement.
System	Breakpoint

**Comment by MatinF** [comment]
Hi tsayman, thanks for the report.

We're aware of the Vector support issue on the finalized MDF. This is because Vector tools do not currently support the ASAM MDF 4.11 extended ID field, CAN_DataFrame.IDE. Instead, a different syntax is used where the CAN ID uses 32 bits instead of 29 bits.

We're looking into how to resolve this best, which may be by updating the raw bus logging format in the next firmware. Doing so resolves the Vector compatibility issue.

Until the compatibility with the finalized MDF files are resolved, we suggest to use the `mdf2asc` converter as this should provide full compatibility with e.g. CANalyzer. 

We'll provide an update here once the `mdf2finalized` compatibility is resolved.

Best,
Martin

**Comment by MatinF** [comment]
Hi again,

A brief update on this: 

We've reviewed the option of changing the MDF4 log file format from the current 29 bit syntax for extended CAN IDs to the alternative 32 bit syntax used by e.g. Vector's tools. 

It is our belief that the 29 bit syntax is the more "correct" syntax to use going forward and we expect that future tools built to support MDF4 will be utilizing this syntax rather than the 32 bit version. 

In addition, we've tested various tools in regards to the support for the 29 bit vs. 32 bit syntax. In this we found that some tools would provide misleading results when loading a 32 bit syntax MDF4 file - e.g. MATLAB's vehicle network toolbox. 

As a result, we've decided to keep the 29 bit syntax for now and we therefore ask that you use the `mdf2asc` converter to load your log files into Vector's CANalyzer/CANape tools. We've updated the documentation accordingly.

At the same time, we'll be working to encourage Vector to support the 29 bit syntax in addition to the 32 bit syntax. Since the 29 bit syntax is part of the official standard and the fact that supporting this should be a minor change, we hope to get this through. If so, we'll inform about this in here.

If any questions, let us know. Closing this for now.

Martin
